visualization/dashboard/src/MLIDashboard/ValidateProperties.ts

```diff
@@ -0,0 +1,225 @@
+import { IExplanationDashboardProps } from "./Interfaces";
+import { IExplanationModelMetadata } from "./IExplanationContext";
+import { localization } from "../Localization/localization";
+
+export class ValidateProperties {
+    private readonly classLength: number;
+    private readonly featureLength: number;
+    public readonly errorStrings: string[] = [];
+    private rowLength: number;
+    constructor(private props: IExplanationDashboardProps, modelMetadata: IExplanationModelMetadata) {
+        this.classLength = modelMetadata.classNames.length;
+        this.featureLength = modelMetadata.featureNames.length;
+        this.validateProps();
+    }
+
+    // Mutates the passed in props arg, removing any properties that are incompatible.
+    private validateProps(): void {
+        if (this.props.trueY) {
+            this.rowLength = this.props.trueY.length;
+        }
+        if (this.props.predictedY) {
+            this.validatePredictedY();
+        }
+        if (this.props.probabilityY) {
+            this.validatePredictProba();
+        }
+        if (this.props.testData) {
+            this.validateTestData();
+        }
+        if (this.props.precomputedExplanations && this.props.precomputedExplanations.localFeatureImportance && this.props.precomputedExplanations.localFeatureImportance.scores) {
+            this.validateLocalExplanation();
+        }
+    }
+
+    private static validatePredictedProbabilities(props: IExplanationDashboardProps, modelMetadata: IExplanationModelMetadata, classLength: number, rowLength?: number): string | undefined {
+        if (props.probabilityY) {
+            if (!Array.isArray(props.probabilityY)) {
+                props.probabilityY = undefined;
+                if (rowLength !== undefined) {
+                    rowLength = 0;
+                }
+                return localization.formatString(localization.ValidationErrors.predictedProbNotArray, rowLength, classLength);
+            }
+            const length = props.probabilityY.length;
+            if (rowLength === undefined) {
+                rowLength = length;
+            }
+            if (length !== rowLength) {
+                return `Inconsistent dimensions. Predicted y has length [${length}], expected [${rowLength}]`;
```

Comment:
shouldn't these be localized?

Reply:
oh, great catch! This was the first one I wrote, before I thought to put it all in a shared localized format. Will fix now!